src/azul/service/responseobjects/dynamo_data_access.py

```diff
@@ -78,10 +84,20 @@ def query(self, **kwargs):
 
         :yields: Results of the query
         """
-        for items, count in self._make_query(**kwargs):
+        for items, _, _ in self._make_query(**kwargs):
             for item in items:
                 yield item
 
+    def iterable_query(self, **kwargs):
+        """
+        Similar to :method:`query` but instead of only yield items, this method
+        will yield result as page.
+
+        :yields: Result page of the query
+        """
+        for items, count, last_evaluated_key in self._make_query(**kwargs):
+            yield ResultPage(last_evaluated_key, iter(items), count)
```

Comment:
I don't understand why items is eagerly converted into an iterator. If it's an iterable why can't it remain an iterable? Then ResultPage could just be a NamedTuple instead of this weird iterator wrapper.

The only thing that would be different in calling code, is that it would have to do `for item in page.items` instead of `for item in page` but that's well worth the simplification.

Reply:
I agree with you. I'll do that.

Reply:
This code was from the experiment I did to find the way to query just one page of results.

Reply:
Summary of changes in related to this:
* `_make_query` is now `make_query`.
* `make_query` now returns `Page`, a named tuple, instead of a normal tuple.
* `iterable_query` has been removed in flavour of `make_query`.